lib/TPP/Dialect/Perf/PerfOps.cpp

```diff
@@ -0,0 +1,49 @@
+//===- PerfOps.cpp - Perf dialect ops ---------------------------*- C++ -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "TPP/Dialect/Perf/PerfOps.h"
+#include "TPP/Dialect/Perf/PerfDialect.h"
+
+#define GET_OP_CLASSES
+#include "TPP/Dialect/Perf/PerfOps.cpp.inc"
+
+using namespace mlir;
+using namespace mlir::perf;
+
+LogicalResult StopTimerOp::verify() {
+  auto timerSrc = getTimer().getDefiningOp();
+  if (!timerSrc || !isa<StartTimerOp>(timerSrc))
+    return emitOpError("invalid timer input");
+
+  int numStopTimers = 0;
+  for (auto user : timerSrc->getUsers()) {
+    if (isa<StopTimerOp>(*user))
+      ++numStopTimers;
+  }
+  if (numStopTimers != 1)
+    return emitOpError("timer stopped multiple times");
+
+  return success();
+}
+
+LogicalResult YieldOp::verify() {
+  // Get the parent operation to check its return values
+  auto benchOp = (*this)->getParentOfType<BenchOp>();
+  if (!benchOp)
+    return emitOpError("invalid parent operation");
+
+  // Check that body results match the yield operands
+  auto types =
```

Comment:
Probably also this logic can be move in the .td file. See for example https://github.com/llvm/llvm-project/blob/78c033b541f00d12b7e1ba2a676b02e022c9beb1/mlir/include/mlir/Dialect/Tensor/IR/TensorOps.td#L1669

Reply:
Yup that worked, thanks.
I moved the result check to `perf.bench` directly and, with the other fix, I removed the `perf.yield` verifier.